node/cmd/guardiand/adminserver.go

```diff
@@ -539,3 +564,106 @@ func (s *nodePrivilegedService) PurgePythNetVaas(ctx context.Context, req *nodev
 		Response: resp,
 	}, nil
 }
+
+func (s *nodePrivilegedService) SignExistingVAA(ctx context.Context, req *nodev1.SignExistingVAARequest) (*nodev1.SignExistingVAAResponse, error) {
+	v, err := vaa.Unmarshal(req.Vaa)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal VAA: %w", err)
+	}
+
+	if req.NewGuardianSetIndex <= v.GuardianSetIndex {
+		return nil, errors.New("new guardian set index must be higher than provided VAA")
+	}
+
+	if s.evmConnector == nil {
+		return nil, errors.New("the node needs to have an Ethereum connection configured to sign existing VAAs")
+	}
+
+	var gs *common.GuardianSet
+	if cachedGs, exists := s.gsCache.Load(v.GuardianSetIndex); exists {
+		gs = cachedGs.(*common.GuardianSet)
+	} else {
+		evmGs, err := s.evmConnector.GetGuardianSet(ctx, v.GuardianSetIndex)
```

Comment:
I think evmConnector could be nil (if line 436 was not satisfied)?

Reply:
See L578